src/Microsoft.ML.Data/Scorers/FeatureContributionCalculatingTransformer.cs

```diff
@@ -41,88 +49,192 @@ namespace Microsoft.ML.Runtime.Data
     /// ]]>
     /// </format>
     /// </example>
-    public sealed class FeatureContributionCalculationTransform
+    public sealed class FeatureContributionCalculatingTransformer : RowToRowTransformerBase
     {
         // Apparently, loader signature is limited in length to 24 characters.
-        internal const string MapperLoaderSignature = "WTFBindable";
-        private const int MaxTopBottom = 1000;
+        internal const string Summary = "For each data point, calculates the contribution of individual features to the model prediction.";
+        internal const string FriendlyName = "Feature Contribution Transform";
+        internal const string LoaderSignature = "FeatureContribution";
 
-        public sealed class Arguments : ScorerArgumentsBase
-        {
-            [Argument(ArgumentType.AtMostOnce, HelpText = "Number of top contributions", SortOrder = 1)]
-            public int Top = 10;
+        private readonly string _featureColumn;
+        private readonly BindableMapper _mapper;
 
-            [Argument(ArgumentType.AtMostOnce, HelpText = "Number of bottom contributions", SortOrder = 2)]
-            public int Bottom = 10;
+        private static VersionInfo GetVersionInfo()
+        {
+            return new VersionInfo(
+                modelSignature: "FCC TRAN",
+                verWrittenCur: 0x00010001, // Initial
+                verReadableCur: 0x00010001,
+                verWeCanReadBack: 0x00010001,
+                loaderSignature: LoaderSignature,
+                loaderAssemblyName: typeof(FeatureContributionCalculatingTransformer).Assembly.FullName);
+        }
 
-            [Argument(ArgumentType.AtMostOnce, HelpText = "Whether or not output of Features contribution should be normalized", ShortName = "norm", SortOrder = 3)]
-            public bool Normalize = true;
+        // TODO documentation
+        public FeatureContributionCalculatingTransformer(IHostEnvironment env, IPredictor predictor, string featureColumn,
+            int top = FeatureContributionCalculatingEstimator.Defaults.Top,
+            int bottom = FeatureContributionCalculatingEstimator.Defaults.Bottom,
+            bool normalize = FeatureContributionCalculatingEstimator.Defaults.Normalize,
+            bool stringigy = FeatureContributionCalculatingEstimator.Defaults.Stringify)
+            : base(Contracts.CheckRef(env, nameof(env)).Register(nameof(FeatureContributionCalculatingTransformer)))
+        {
+            Host.CheckValue(predictor, nameof(predictor));
+            Host.CheckNonEmpty(featureColumn, nameof(featureColumn));
 
-            [Argument(ArgumentType.AtMostOnce, HelpText = "Whether or not output of Features contribution in string key-value format", ShortName = "str", SortOrder = 4)]
-            public bool Stringify = false;
+            var pred = predictor as IFeatureContributionMapper;
+            Host.CheckParam(pred != null, nameof(predictor), "Predictor doesn't support getting feature contributions");
```

Comment:
It seems wrong to have a runtime exception. Is there anyway we can have this as a compile-time error?